### ISSUE TITLE
Can't connect to storefront after build successfull

### ISSUE THREAD

**Issue by mrtuvn** [magento-research/pwa-studio#583]
## This issue is for the following packages:

[X] `venia-concept`
[ ] `pwa-buildpack`
[ ] `peregrine`
[ ] `pwa-devdocs`
[ ] `upward-js`
[ ] `upward-spec`

## This issue is a:

[ ] Bug
[ ] Feature suggestion
[ ] Documentation issue
[X] Other (Problem with setup. This maybe common case. Server host in guest machine docker with installed apache server with node and npm)

### Environment

| Question                    |    Answer     |
| --------------------------- | :-----------: |
| Magento version             | 2.3 stable|
| Operating System + version  | Ubuntu 18.10 |
| node.js version (`node -v`) | 10.14.0 |
| npm version (`npm -v`)      | 6.4.1 |

## Description

I got issue unable connect storefront in browser. Used firefox. Build log successfull
I have added domain to my hosts file both guest and host (ubuntu) enviroment. Used docker ce
Docker version 18.09.0
My docker enviroment used from this repo
https://github.com/fballiano/docker-magento2
Only mount files from host and edit with VSCode

## Expected result:

Can view storefront of venia in browser

## Other problems still remain
1 - Updated file .env in pwa-studo-folder/packages/venia-concept/.env with my own url local magento dev but after build pwa-studio still render url with path like this https://magento-venia-concept-fznpj.local.pwadev
2 - In order to build successfull i must delete folders then rebuild again
.config/devcert/certificate-authority/*
.config/devcert/domains/*

**Comment by ennostuurman** [comment]
@mrtuvn I am running PWA Studio with local Magento instance on Docker as backend on Ubuntu Bionic without issues . So maybe I can help you getting up and running.

Just to be sure, are you running PWA Studio from Ubuntu shell directly or from within your Docker container?

**Comment by mrtuvn** [comment]
@ennostuurman i am running pwa studio inside docker container use docker exec command to connect
Problem may come from docker can't forward port to host machine

**Comment by ennostuurman** [comment]
@mrtuvn yep that could be the issue. But why run PWA Studio inside Docker when you can run it directly in the shell? That's how I use it and is probably less error prone with regard to port and network configurations.

So:

- install PWA Studio code base in your home folder
- change MAGENTO_BACKEND_URL in venia-concept .env file to the Magento store url of your Docker instance
- startup Docker
- run the npm run build and npm run watch:venia from PWA Studio root
- now your venia storefront should be able to proxy to the Docker instance (note: works best with Venia sample data installed because of some hardcoded dependencies, i.e. configurable options).

In case of troubles try PWA Studio with default configured remote M2.3 instance first to see if it works, then try to connect to your Docker Magento instance just by changing the .env file as described above.

**Comment by mrtuvn** [comment]
thanks i will try it again

**Comment by ennostuurman** [comment]
hey @mrtuvn did you manage to get PWA Studio running?

**Comment by sharkySharks** [comment]
Work will be tracked through #734 .

**Comment by mrtuvn** [comment]
@ennostuurman today i back to this project with new build with yarn. But i got issue with graphql 
{"errors":[{"message":"Internal server error","category":"internal"}]}

sampledata installed
used 2.3 branch git repo magento for instance base 
all requirements installed

Any suggestions with docker enviroments ? @sharkySharks

**Comment by mrtuvn** [comment]
How to resolve this issue
![screenshot from 2019-02-15 10-06-58](https://user-images.githubusercontent.com/1908873/52832049-8063c680-3109-11e9-9ce6-31e3dbbb483f.png)

**Comment by mrtuvn** [comment]
seem default mode magento not work with pwa. So i set to developer mode it's seem work now
But still can't reproduce pwa server running in my local machine with custom port like this article
https://magento-research.github.io/pwa-studio/venia-pwa-concept/setup/

**Comment by mrtuvn** [comment]
Ok seem everything works now. But image can't load via pwadevserver. This is problem exist or fixed ?

**Comment by sharkySharks** [comment]
Hi @mrtuvn glad you got it working. Glad others are using docker for their setup as well. I have been working on a docker setup script that you can find in my [pr](https://github.com/magento-research/pwa-studio/pull/859) that you can run locally to get the dev server up and running. AFAIK, the pwadevserver needed some tweaks in order to get it to work with docker properly. If you have things working on your own then great! If you want to try my setup script I would appreciate any feedback for improvements.

**Comment by tjwiebell** [comment]
Image loading is a known issue if you're not using `/pub` as your document root. There is a proposed fix in #567 that will grab the correct media path from the Magento backend, so no changes are needed in Venia. Steps to resolve in the meantime:

`packages/venia-concept/.env`
```
MAGENTO_BACKEND_MEDIA_PATH_PRODUCT="/pub/media/catalog/product"
MAGENTO_BACKEND_MEDIA_PATH_CATEGORY="/pub/media/catalog/category"
```

`packages/venia-concept/venia-upward.yml`
```yaml
# change the proxy pattern to
- matches: request.url.pathname
  pattern: '^/(graphql|rest|media|pub)'
  use: proxy
```
